registration-processor/registration-processor-common-camel-bridge/src/test/java/io/mosip/registration/processor/camel/bridge/WorkflowCommandPredicateTest.java

```diff
@@ -0,0 +1,1757 @@
+
+			}
+
+			@Override
+			public void addService(Object object, boolean stopOnShutdown, boolean forceStart) throws Exception {
+				// TODO Auto-generated method stub
+
+			}
+
+			@Override
+			public void addService(Object object, boolean stopOnShutdown) throws Exception {
+				// TODO Auto-generated method stub
+
+			}
+
+			@Override
+			public void addService(Object object) throws Exception {
+				// TODO Auto-generated method stub
+
+			}
+
+			@Override
+			public void addRoutes(RoutesBuilder builder) throws Exception {
+				// TODO Auto-generated method stub
+
+			}
+
+			@Override
+			public void addRoutePolicyFactory(RoutePolicyFactory routePolicyFactory) {
+				// TODO Auto-generated method stub
+
+			}
+
+			@Override
+			public void addRouteDefinitions(Collection<RouteDefinition> routeDefinitions) throws Exception {
+				// TODO Auto-generated method stub
+
+			}
+
+			@Override
+			public void addRouteDefinition(RouteDefinition routeDefinition) throws Exception {
+				// TODO Auto-generated method stub
+
+			}
+
+			@Override
+			public void addRestDefinitions(Collection<RestDefinition> restDefinitions) throws Exception {
+				// TODO Auto-generated method stub
+
+			}
+
+			@Override
+			public void addRestConfiguration(RestConfiguration restConfiguration) {
+				// TODO Auto-generated method stub
+
+			}
+
+			@Override
+			public void addRegisterEndpointCallback(EndpointStrategy strategy) {
+				// TODO Auto-generated method stub
+
+			}
+
+			@Override
+			public void addLogListener(LogListener listener) {
+				// TODO Auto-generated method stub
+
+			}
+
+			@Override
+			public void addLifecycleStrategy(LifecycleStrategy lifecycleStrategy) {
+				// TODO Auto-generated method stub
+
+			}
+
+			@Override
+			public void addInterceptStrategy(InterceptStrategy interceptStrategy) {
+				// TODO Auto-generated method stub
+
+			}
+
+			@Override
+			public void addHystrixConfiguration(String id, HystrixConfigurationDefinition configuration) {
+				// TODO Auto-generated method stub
+
+			}
+
+			@Override
+			public Endpoint addEndpoint(String uri, Endpoint endpoint) throws Exception {
+				// TODO Auto-generated method stub
+				return null;
+			}
+
+			@Override
+			public void addComponent(String componentName, Component component) {
+				// TODO Auto-generated method stub
+
+			}
+
+			@Override
+			public <T extends CamelContext> T adapt(Class<T> type) {
+				// TODO Auto-generated method stub
+				return null;
+			}
+		});
+	}
+
+	@SuppressWarnings("unchecked")
+	@Test
+	public void testRoutePredicatePositiveForCompleteAsProcessed() throws Exception {
+		MessageDTO messageDTO = new MessageDTO();
+		messageDTO.setRid("10002100741000120201231071308");
+		exchange.getMessage().setBody(objectMapper.writeValueAsString(messageDTO));
+		exchange.getMessage().setHeader(Exchange.INTERCEPTED_ENDPOINT, "workflow-cmd://complete-as-processed");
+		assertTrue(workflowCommandPredicate.matches(exchange));

```

Comment:
For all the below cases you have to check with the exchange message is updated appropriately only checking matches is not good enough

Reply:
fixed